mirascope/core/base/toolkit.py

```diff
@@ -0,0 +1,92 @@
+"""The module for defining the toolkit class for LLM call tools."""
+
+from __future__ import annotations
+
+import inspect
+from abc import ABC
+from typing import Callable, ClassVar, NamedTuple
+
+from pydantic import BaseModel, ConfigDict
+from typing_extensions import ParamSpec, Concatenate
+
+from . import BaseTool
+from ._utils import convert_function_to_base_tool, get_template_variables
+
+_TOOLKIT_TOOL_METHOD_MARKER: str = "__toolkit_tool_method__"
+
+_namespaces: set[str] = set()
+
+P = ParamSpec("P")
+
+
+def toolkit_tool(
+    method: Callable[Concatenate[BaseToolKit, P], str],
+) -> Callable[Concatenate[BaseToolKit, P], str]:
+    # Mark the method as a toolkit tool
+    setattr(method, _TOOLKIT_TOOL_METHOD_MARKER, True)
+
+    return method
+
+
+class ToolKitToolMethod(NamedTuple):
+    method: Callable[..., str]
+    template_vars: list[str]
+    template: str
+
+
+class BaseToolKit(BaseModel, ABC):
+    """A class for defining tools for LLM call tools."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    _toolkit_tool_methods: ClassVar[list[ToolKitToolMethod]]
+    _namespace: ClassVar[str | None] = None
```

Comment:
what do you think about using `__namespace__`?

Reply:
Looks good. I will change it soon.

Reply:
Done!